composition/string_fragment.go

```diff
@@ -5,13 +5,29 @@ import (
 )
 
 // StringFragment is a simple template based representation of a fragment.
-type StringFragment string
+type StringFragment struct {
```

Comment:
It's ok for me.
It will be needed if more metadata will be associated with a fragment.